src/api/cvc4cpp.h

```diff
@@ -590,6 +698,43 @@ class CVC4_PUBLIC Term
    */
   bool isNull() const;
 
+  /**
+   * @return true iff this Term has an operator
+   */
+  bool hasOperator() const;
+
+  /**
+   * @return true iff this Term has a builtin operator
+   *
+   * in this case, getKind() will return the operator
+   */
+  bool hasBuiltinOperator() const;
+
+  /**
+   * @return true iff this Term has an indexed operator
+   */
+  bool hasOpTermOperator() const;
+
+  /**
+   * @return true iff this Term is the result of
+   *         applying an uninterpreted function
+   */
+  bool hasUFOperator() const;
```

Comment:
The name of this function does not really imply what it actually does. Can we maybe find a better name?